Sources/Controllers/2.0/OARearrangeCustomFiltersViewController.mm

```diff
@@ -0,0 +1,455 @@
+    NSIndexPath *targetPath = [NSIndexPath indexPathForRow:order inSection:kAllFiltersSection];
+    [CATransaction begin];
+    [CATransaction setCompletionBlock:^{
+        [_tableView reloadData];
+    }];
+    [_hiddenFiltersItems removeObjectAtIndex:indexPath.row];
+    [_filtersItems insertObject:filterItem atIndex:order];
+    [_hiddenFiltersKeys removeObject:filterItem.filter.filterId];
+    filterItem.filter.isActive = YES;
+    [_tableView beginUpdates];
+    [_tableView moveRowAtIndexPath:indexPath toIndexPath:targetPath];
+    [_tableView endUpdates];
+    [CATransaction commit];
+}
+
+- (void)updateFiltersIndexes
+{
+    for (int i = 0; i < _filtersItems.count; i++)
+        _filtersItems[i].order = i;
+}
+
+- (void)showChangesAlert
+{
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle:nil message:OALocalizedString(@"osm_editing_lost_changes_title") preferredStyle:UIAlertControllerStyleAlert];
+    [alert addAction:[UIAlertAction actionWithTitle:OALocalizedString(@"shared_string_exit") style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+        [self.navigationController popViewControllerAnimated:YES];
+    }]];
+    [alert addAction:[UIAlertAction actionWithTitle:OALocalizedString(@"shared_string_cancel") style:UIAlertActionStyleCancel handler:nil]];
+    [self presentViewController:alert animated:YES completion:nil];
+}
+
+#pragma mark - UITableViewDataSource
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath
+{
+    NSString *cellType = indexPath.section == kActionsSection ? kButtonRightIconCell : kHideButtonCell;
+    if ([cellType isEqualToString:kHideButtonCell])
+    {
+        BOOL isAllFilters = indexPath.section == kAllFiltersSection;
+        OAPOIUIFilter *filter = isAllFilters ? _filtersItems[indexPath.row].filter : _hiddenFiltersItems[indexPath.row].filter;
+        OADeleteButtonTableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:kHideButtonCell];
+        if (cell == nil)
+        {
+            NSArray *nib = [[NSBundle mainBundle] loadNibNamed:kHideButtonCell owner:self options:nil];
+            cell = (OADeleteButtonTableViewCell *) nib[0];
+            cell.selectionStyle = UITableViewCellSelectionStyleNone;
+            cell.separatorInset = UIEdgeInsetsMake(0.0, 58.0, 0.0, 0.0);
+        }
+        if (cell)
+        {
+            cell.titleLabel.text = filter.name;
+            UIImage *poiIcon = [UIImage templateImageNamed:filter.getIconId];
+            cell.iconImageView.image = poiIcon ? poiIcon : [UIImage templateImageNamed:@"ic_custom_user"];
+            NSString *imageName = isAllFilters ? @"ic_custom_delete" : @"ic_custom_plus";
+            [cell.deleteButton setImage:[UIImage imageNamed:imageName] forState:UIControlStateNormal];
+            [cell.deleteButton setUserInteractionEnabled:YES];
```

Comment:
If for some reason the user interaction is disabled on deleteButton by default, either figure out why is it disabled in the xib and enable it there, or if it is correct behavior for the cell, enable it in the first if clause (when the cell is created for the first time). If, however, it is enabled in the xib, remove this line completely.